core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceRepository.scala

```diff
@@ -30,6 +30,11 @@ class WorkspaceRepository(dataSource: SlickDataSource) {
       access.workspaceQuery.updateState(workspaceId, state)
     }
 
+  def setFailedState(workspaceId: UUID, state: WorkspaceState, message: String): Future[Int] =
```

Comment:
I went this route over adding an optional `message` param to `updateState` with the hope that it would make it clear to us in the future that you need to include an error message when you're setting a workspace to a `*Failed` state.